app/models/supported_ssg.rb

```diff
@@ -74,9 +74,28 @@ def available_upstream
       end
     end
 
+    # Multilevel map of latest supported SSG for OS major and minor version
+    def latest_map
+      @latest_map ||= begin
+        all.group_by(&:os_major_version)
+           .transform_values do |major_ssgs|
+             major_ssgs
+               .group_by(&:os_minor_version)
+               .transform_values do |minor_ssgs|
+                 minor_ssgs.max_by(&:comparable_version)
+               end.freeze
+           end.freeze
+      end
+    end
+
+    def latest_map!
```

Comment:
I'd rather have this as:
```ruby
def latest_map
  @latest_map ||= build_latest_map
end

def build_latest_map
  ...
end
```

Reply:
Updated. I've kept the `latest_map!`.

Reply:
:+1:

Reply:
Wait, are you using the `latest_map!` only in tests? Because the purpose of `build_latest_map` would be to call that one in the tests :thinking:

Reply:
Yes, only used in tests atm. There are some integration tests that need to clear the cached value (instance variable). It can be `clear_latest_map!` (which I guess its the same) or the tests can mangle with the class instance variable.

Reply:
Oh, you can just do `SupportedSsg.instance_variable_set(:@latest_map, nil)` which would clear the thing from the test and then you don't need any extra method.

Reply:
Yep, I've used it before I've decided to do a public method that would also clear it. Ok, I'll roll it back...